### PR TITLE
Aggrid

### PR DESCRIPTION
**Pull request by spryor**
# Type of change? 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation
Migrate PcsGrid from RM version 1 to version 2

### CODE REVIEW

File: src/components/pages/devices/devicesGrid/devicesConfig.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+export const checkboxParams = {
+  headerCheckboxSelection: true,
+  headerCheckboxSelectionFilteredOnly: true,
+  checkboxSelection: true
+};
+
+/** A collection of column definitions for the devices grid */
+export const deviceColumnDefs = {
+  id: {
+    headerName: 'Device Name',
```
Comment (ppathan) [inline src/components/pages/devices/devicesGrid/devicesConfig.js]: Do we need to localize this? Should this be hard coded? Same comment for other places

File: src/components/shared/cellRenderers/isSimulatedRenderer/isSimulatedRenderer.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import { Svg } from 'components/shared/svg/svg';
+import { svgs } from 'utilities';
+
+import '../cellRenderer.css'
+
+export const IsSimulatedRenderer = ({ value }) => (
+  <div className="pcs-renderer-cell highlight">
+    <Svg path={value ? svgs.simulatedDevice : svgs.physicalDevice} className="pcs-renderer-icon" />
+    <div className="pcs-renderer-text">
+      { value ? 'YES' : 'NO' }
```
Comment (ppathan) [inline src/components/shared/cellRenderers/isSimulatedRenderer/isSimulatedRenderer.js]: Does this need to be localized?

File: src/components/shared/pcsGrid/pcsGrid.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  };
+
+  /** Computes the CSS classes to apply to each row, including soft select */
+  getRowClass = params => {
+    let rowClasses = '';
+    if (isFunc(this.props.getSoftSelectId)) {
+      rowClasses = this.props.getSoftSelectId(params.data) === this.props.softSelectId ? 'pcs-row-soft-selected' : '';
+    }
+    if (isFunc(this.props.getRowClass)) {
+      rowClasses += ` ${this.props.getRowClass(params)}`;
```
Comment (ppathan) [inline src/components/shared/pcsGrid/pcsGrid.js]: Is space needed between **` $**

File: src/components/shared/pcsGrid/pcsGrid.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    };
+
+    this.subscriptions = [];
+    this.clickStream = new Rx.Subject();
+  }
+
+  componentDidMount() {
+    this.subscriptions.push(
+      this.clickStream
+        .debounceTime(180)
```
Comment (ppathan) [inline src/components/shared/pcsGrid/pcsGrid.js]: Make 180 a constant variable

File: src/components/shared/pcsGrid/pcsGridConfig.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft. All rights reserved.
+/* This file contains default values useful for creating PcsGrid */
+
+/** The default value for PcsGrid cells that are empty */
+export const EMPTY_FIELD_VAL = '---';
+
+/** The default formatting for dates in the PcsGrid */
+export const DEFAULT_TIME_FORMAT = 'hh:mm:ss MM.DD.YYYY';
```
Comment (ppathan) [inline src/components/shared/pcsGrid/pcsGridConfig.js]: How are we overriding this format?

Comment (spryor) [reply src/components/shared/pcsGrid/pcsGridConfig.js]: Currently, it isn't being overridden. This should be replaced later by a component that handles rendering timestamps.

File: src/components/shared/pcsGrid/pcsGrid.js (hunk 4 above)
Comment (spryor) [reply src/components/shared/pcsGrid/pcsGrid.js]: Done

**Pull request merged**